### PR TITLE
Pod events stored with the Run object

### PR DESCRIPTION
**Pull request by ujjwalsfix**
- Fetched and deduplicated
- Fetching interval changed to 20 second vs 5 seconds
- UI code also updated

### CODE REVIEW

File: worker/status_worker.go [hunk 1]
```diff
@@ -139,7 +139,8 @@ func (sw *statusWorker) processRun(run state.Run) {
 			if updatedRun.MaxMemoryUsed != run.MaxMemoryUsed ||
 				updatedRun.MaxCpuUsed != run.MaxCpuUsed ||
 				updatedRun.Cpu != run.Cpu ||
-				updatedRun.Memory != run.Memory {
+				updatedRun.Memory != run.Memory ||
+				updatedRun.PodEvents != run.PodEvents {
```
Comment (psunnn) [inline worker/status_worker.go]: out of curiosity, is the `!=` operator sufficient for comparing list of structs??

File: state/pg_state_manager.go [hunk 2]
```diff
@@ -865,6 +864,21 @@ func (e EnvList) Value() (driver.Value, error) {
 	return res, nil
 }
 
+// Scan from db
+func (e *PodEvents) Scan(value interface{}) error {
```
Comment (psunnn) [inline state/pg_state_manager.go]: ah so this is how you do it 👍

File: execution/engine/eks_engine.go [hunk 3]
```diff
@@ -389,12 +388,43 @@ func (ee *EKSExecutionEngine) FetchUpdateStatus(run state.Run) (state.Run, error
...
+	events, err := ee.GetEvents(run)
+
+	if err == nil && len(events.PodEvents) > 0 {
+		newEvents := events.PodEvents
+		if run.PodEvents != nil && len(*run.PodEvents) > 0 {
+			priorEvents := *run.PodEvents
+			for _, newEvent := range newEvents {
+				unseen := true
+				for _, priorEvent := range priorEvents {
+					if newEvent.Reason == priorEvent.Reason &&
```
Comment (psunnn) [inline execution/engine/eks_engine.go]: perhaps extract this into a `arePodEventsEqual` function for your sanity later on

Comment (ujjwalsfix) [reply execution/engine/eks_engine.go]: good point, fixed

**Pull request merged**